### PR TITLE
Inference Rules Updates for multiple Nodes

### PR DESCRIPTION
**Pull request by liorkhe**
Clarification of Network inference rules for supporting SUT containing Multiple Nodes.

### CODE REVIEW

File: inference_rules.adoc [hunk 1]
```diff
@@ -445,8 +445,10 @@ Non-conforming network submission should be submitted to Open category, under th
 * The QDL is not allowed to pad the data in queries.
 * The QDL is not allowed to cache queries or responses.
 * The QDL is implementing the network function of the LoadGen Node towards the SUT node and handles the required processing. E.G. padding of the payload as required by the network protocol.
+* The QDL should reflect a single SUT to the LoadGen. LoadGen Operates with a single SUT.
```
Comment (DilipSequeira) [inline inference_rules.adoc:448]:
```suggestion
* The QDL should reflect a single SUT to the LoadGen. LoadGen operates with a single SUT. 
```

File: inference_rules.adoc [hunk 2]
```diff
@@ -445,8 +445,10 @@ Non-conforming network submission should be submitted to Open category, under th
 * The QDL is not allowed to pad the data in queries.
 * The QDL is not allowed to cache queries or responses.
 * The QDL is implementing the network function of the LoadGen Node towards the SUT node and handles the required processing. E.G. padding of the payload as required by the network protocol.
+* The QDL should reflect a single SUT to the LoadGen. LoadGen Operates with a single SUT. 
 * The Name method's return value must contain the substring "Network SUT".
 * The Name method's implementation must include at least one round trip over the network. The Name method must not return until the round trip is complete.
+* The QDL should serially get the names from all the Nodes of the SUT, aggregate the response in the Name Method after getting response from all Nodes. Each Node must have a unique Name.
```
Comment (DilipSequeira) [inline inference_rules.adoc:451]:
```suggestion
* The QDL should serially get the names from all the Nodes of the SUT and aggregate the responses in the Name Method after getting response from all Nodes. Each Node must have a unique Name.   
```

File: inference_rules.adoc [hunk 3]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
...
-The SUT and QDL must run on two physically separated and distinct systems.
+The SUT and QDL must run on two physically separated and distinct systems. The SUT can contain multiple Nodes.
 
 Fabric and protocol must be reported in the submission metadata. Submission metadata must be sufficient to determine OSI layers one through four of the submission's network stack.
 
 ==== SUT over the Network and Setup Constraints
 
 * SUT parameters and configuration must be uniquely and specifically named in the submission results.
-* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like switch or load balancer should be considered part of the SUT. 
+* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.
```
Comment (DilipSequeira) [inline inference_rules.adoc:480]:
```suggestion
* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like a switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.
```

File: inference_rules.adoc [hunk 4]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
...
 * Caching/Storing of the queries and inference data or responses for further use at the SUT is disallowed. It is allowed to cache/store other data like Neural Network weights or Neural Network executable.
 * SUT can do the required pre-processing of the data, e.g. Batching, Padding, processing of the requests (precision, data layout), compression, decompression. SUT can do the required post processing functions e.g. gather, reduction or ArgMax. 
-* Network interface characteristics must be added to the report, both at the SUT side and at the LoadGen system side, including networking components, and name any other components through which data passes between LoadGen and the SUT for reproducibility.
-* It is required to add a system diagram that shows how the components between the LoadGen node and the SUT node are connected and to add text that explains the diagram.
+* Network interface characteristics must be added to the report, both at the SUT side, for every Node, and at the LoadGen system side, including networking components, and name any other components through which data passes between LoadGen and the SUT for reproducibility.
```
Comment (DilipSequeira) [inline inference_rules.adoc:484]:
```suggestion
* Network interface characteristics must be added to the report, both at the SUT side for every Node, and at the LoadGen system side, including networking components. Every other component through which data passes between LoadGen and the SUT must also be described for reproducibility purposes.
```

File: inference_rules.adoc [hunk 5]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
...
+* It is required to add a system diagram that shows how the components between the LoadGen node and the SUT nodes are connected and to add text that explains the diagram.
```
Comment (DilipSequeira) [inline inference_rules.adoc:485]:
```suggestion
* A system diagram must be included in the submission that shows how the components between the LoadGen node and the SUT nodes are connected, accompanied by any text necessary for another submitter to understand the diagram.
```

**Review by DilipSequeira — COMMENTED** [review]
minor copy edit suggestions.

File: inference_rules.adoc (hunk 1 above)
Comment (liorkhe) [reply inference_rules.adoc]: changed

File: inference_rules.adoc (hunk 2 above)
Comment (liorkhe) [reply inference_rules.adoc]: changed

File: inference_rules.adoc (hunk 3 above)
Comment (liorkhe) [reply inference_rules.adoc]: changed

File: inference_rules.adoc (hunk 4 above)
Comment (liorkhe) [reply inference_rules.adoc]: changed

File: inference_rules.adoc (hunk 5 above)
Comment (liorkhe) [reply inference_rules.adoc]: changed

File: inference_rules.adoc [hunk 6]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
 
 Additionally, any interconnect not listed in the permitted list is forbidden unless clearance is first obtained from the MLPerf Inference WG.
 
-The SUT and QDL must run on two physically separated and distinct systems.
+The SUT and QDL must run on two physically separated and distinct systems. The SUT can contain multiple Nodes.
```
Comment (DilipSequeira) [inline inference_rules.adoc:473]:
```suggestion
The SUT and QDL must run on physically separate and distinct systems. The SUT can contain multiple Nodes.
```

File: inference_rules.adoc [hunk 7]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
...
 
 Fabric and protocol must be reported in the submission metadata. Submission metadata must be sufficient to determine OSI layers one through four of the submission's network stack.
 
 ==== SUT over the Network and Setup Constraints
 
 * SUT parameters and configuration must be uniquely and specifically named in the submission results.
-* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like switch or load balancer should be considered part of the SUT. 
+* Everything outside the LoadGen node should be considered as part of the SUT, for instance for counting power and latency. As an example, components outside the nodes like a switch or load balancer should be considered part of the SUT. The SUT can contain multiple Nodes.
```
Comment (DilipSequeira) [inline inference_rules.adoc:480]: Just for "don't repeat yourself" reasons I suggest we just say that the SUT can contain multiple nodes one time.

File: inference_rules.adoc [hunk 8]
```diff
@@ -468,19 +470,19 @@ Examples of forbidden bus interconnect include: PCIe/CXL/CCIX, Hypertransport, N
...
 * Caching/Storing of the queries and inference data or responses for further use at the SUT is disallowed. It is allowed to cache/store other data like Neural Network weights or Neural Network executable.
 * SUT can do the required pre-processing of the data, e.g. Batching, Padding, processing of the requests (precision, data layout), compression, decompression. SUT can do the required post processing functions e.g. gather, reduction or ArgMax. 
-* Network interface characteristics must be added to the report, both at the SUT side and at the LoadGen system side, including networking components, and name any other components through which data passes between LoadGen and the SUT for reproducibility.
-* It is required to add a system diagram that shows how the components between the LoadGen node and the SUT node are connected and to add text that explains the diagram.
+* Network interface characteristics must be added to the report, both at the SUT side for every Node, and at the LoadGen system side, including networking components.  Every other component through which data passes between LoadGen and the SUT must also be described for reproducibility purposes.
```
Comment (DilipSequeira) [inline inference_rules.adoc:484]:
```suggestion
* The report must contain network interface characteristics for both the Loadgen and SUT systems, and every other component through which data passes between Loadgen and SUT. The information must be sufficient for reproducibility.
```

File: inference_rules.adoc [hunk 9]
```diff
@@ -445,8 +445,10 @@ Non-conforming network submission should be submitted to Open category, under th
 * The QDL is not allowed to pad the data in queries.
 * The QDL is not allowed to cache queries or responses.
 * The QDL is implementing the network function of the LoadGen Node towards the SUT node and handles the required processing. E.G. padding of the payload as required by the network protocol.
+* The QDL should reflect a single SUT to the LoadGen. LoadGen operates with a single SUT. 
 * The Name method's return value must contain the substring "Network SUT".
 * The Name method's implementation must include at least one round trip over the network. The Name method must not return until the round trip is complete.
+* The QDL should serially get the names from all the Nodes of the SUT and aggregate the responses in the Name Method after getting response from all Nodes. Each Node must have a unique Name.
```
Comment (DilipSequeira) [inline inference_rules.adoc:451]:
```suggestion
* The QDL must query each SUT Node for its name and aggregate the responses in the Name Method. Each SUT Node must have a unique name.   
```

Comment (liorkhe) [reply inference_rules.adoc:451]: One clarification please - the word serially is not appearing in your text. Do we care if in multiple nodes we query in parallel? Do we consider the latency accumulation as an additional information or just want to see it is Network > local?

**Comment by DilipSequeira** [comment]
My recollection is that some people did indeed want to see the latency from the SUT nodes as well as the names (I don't have a strong position on this.) If we want that, then we should have QDL report that information to loadgen, and we should require it in the rules.

Even if we are reporting the latency of the individual SUT nodes, I don't see any advantage in querying them serially.

File: inference_rules.adoc (hunk 6 above)
Comment (liorkhe) [reply inference_rules.adoc:473]: thanks

File: inference_rules.adoc (hunk 7 above)
Comment (liorkhe) [reply inference_rules.adoc:480]: OK. I will delete it here.

File: inference_rules.adoc (hunk 8 above)
Comment (liorkhe) [reply inference_rules.adoc:484]: thanks

File: inference_rules.adoc (hunk 9 above)
Comment (liorkhe) [reply inference_rules.adoc:451]: OK I accept your suggestion

**Review by rnaidu02 — APPROVED** [review]

**Pull request merged**